i3_agenda/API.py

```diff
@@ -0,0 +1,131 @@
+
+import os
+import re
+from typing import List
+import datetime
+import pickle
+
+from google.auth.transport.requests import Request
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build, Resource
+
+from event import Event
+from event_utils import get_event_time, is_allday
+from config import CONF_DIR, URL_REGEX
+
+SCOPES = ["https://www.googleapis.com/auth/calendar.readonly"]
+TMP_TOKEN = f"{CONF_DIR}/i3agenda_google_token.pickle"
+
+
+def connect(credspath: str) -> Resource:
```

Comment:
long function, could probably be separated into 2-3 functions

Reply:
Now I separated it into more functions.